8265426: Update java.security to use instanceof pattern variable

pconcannon opened the pull request:
Hi,

Could someone please review my code for updating the code in the `java.security` package to make use of the `instanceof` pattern variable?

Kind regards,
Patrick

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8265426](https://bugs.openjdk.java.net/browse/JDK-8265426): Update java.security to use instanceof pattern variable


### Reviewers
 * [Roger Riggs](https://openjdk.java.net/census#rriggs) (@RogerRiggs - **Reviewer**)
 * [Weijun Wang](https://openjdk.java.net/census#weijun) (@wangweij - **Reviewer**)
 * [Daniel Fuchs](https://openjdk.java.net/census#dfuchs) (@dfuch - **Reviewer**)


### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/3687/head:pull/3687` \
`$ git checkout pull/3687`

Update a local copy of the PR: \
`$ git checkout pull/3687` \
`$ git pull https://git.openjdk.java.net/jdk pull/3687/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 3687`

View PR using the GUI difftool: \
`$ git pr show -t 3687`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/3687.diff">https://git.openjdk.java.net/jdk/pull/3687.diff</a>

</details>

jespersm commented on src/java.base/share/classes/java/security/Provider.java:
```diff
@@ -2005,16 +2000,16 @@ public boolean supportsParameter(Object parameter) {
                 return true;
             }
-            if (cap.supportsParameter == false) {
+            if (!(cap.supportsParameter)) {
```
Why the extra parens?

pconcannon replied:
Parenthesis removed. See 9e9f9fb

jespersm commented on src/java.base/share/classes/java/security/Provider.java:
```diff
@@ -2005,16 +2000,16 @@ public boolean supportsParameter(Object parameter) {
                     ("Parameter must be instanceof Key for engine " + type);
             }
-            if (hasKeyAttributes() == false) {
+            if (!(hasKeyAttributes())) {
```
Also here: Why the parens?

pconcannon replied:
Parenthesis removed. See 9e9f9fb

jespersm reviewed (commented):
Changes look good, except for the unneeded parenthesis?

(but I'm not a committer, so it's less useful)

RogerRiggs commented on src/java.base/share/classes/java/security/CodeSource.java:
```diff
@@ -157,22 +157,20 @@ public boolean equals(Object obj) {
         // certs must match
-        return matchCerts(cs, true);
+        return matchCerts(other, true);
     }
```
Can this (160-173) be collapsed to:

return (obj instanceof CodeSource cs) && Objects.equals(location, other.location) && matchCerts(cs, true)

pconcannon replied:
Code refactored as suggested. See 9e9f9fb

wangweij commented:
Two comments:

1. Why not reuse the existing variable name (Ex: `t` in `Type t = (Type)obj`) as much as possible to avoid unnecessary renames?
2. I'm not sure if modifying argument name in a public API is a good idea. This leads to unnecessary javadoc changes.

wangweij reviewed (commented):
Two comments:

1. Why not reuse the existing variable name (Ex: `t` in `Type t = (Type)obj`) as much as possible to avoid unnecessary renames?
2. I'm not sure if modifying argument name in a public API is a good idea. This leads to unnecessary javadoc changes.

Punikekk commented on src/java.base/share/classes/java/security/Identity.java:
```diff
@@ -327,24 +327,23 @@ public void removeCertificate(Certificate certificate)
                 return true;
             } else {
-                return identityEquals(i);
+                return identityEquals(other);
             }
         }
         return false;
```
```suggestion
        return obj instanceof Identity other
                && (this.fullName().equals(other.fullName())
                || identityEquals(other));
```

pconcannon replied:
Hi Punikekk, thanks for your comment. I've refactored as suggested. See cbf2841

Punikekk commented on src/java.base/share/classes/java/security/Identity.java:
```diff
@@ -327,24 +327,23 @@ public void removeCertificate(Certificate certificate)
-        if (identity instanceof Identity) {
-            Identity i = (Identity)identity;
-            if (this.fullName().equals(i.fullName())) {
+        if (obj instanceof Identity other) {
+            if (this.fullName().equals(other.fullName())) {
                 return true;
             } else {
```
```suggestion
```

pconcannon replied:
Not really sure what you are suggesting here. Could you clarify?

Punikekk reviewed (commented):
<!-- Start of TOU acceptance message header -->
Hi @Punikekk, thanks for making a comment in an OpenJDK project!

All comments and discussions in the OpenJDK Community must be made available under the OpenJDK [Terms of Use](https://openjdk.java.net/legal/tou/). If you already are an OpenJDK [Author](https://openjdk.java.net/bylaws#author), [Committer](https://openjdk.java.net/bylaws#committer) or [Reviewer](https://openjdk.java.net/bylaws#reviewer), please click [here](https://bugs.openjdk.java.net/secure/CreateIssue.jspa?pid=11300&issuetype=1) to open a new issue so that we can record that fact. Please Use "Add GitHub user Punikekk for the summary.

If you are not an OpenJDK Author, Committer or Reviewer, simply check the box below to accept the OpenJDK Terms of Use for your comments.
<!-- End of TOU acceptance message header -->
 - [ ]  I agree to the [OpenJDK Terms of Use](https://openjdk.java.net/legal/tou/) for all comments I make in a project in the [OpenJDK GitHub organization](https://github.com/openjdk).
<!-- Start of TOU acceptance message footer -->
Your comment will be automatically restored once you have accepted the OpenJDK [Terms of Use](https://openjdk.java.net/legal/tou/).
<!-- End of TOU acceptance message footer -->

<!-- Original message to be restored: 

little+simplification+
-->

pconcannon commented:
> Two comments:
> 
>     1. Why not reuse the existing variable name (Ex: `t` in `Type t = (Type)obj`) as much as possible to avoid unnecessary renames?
> 
>     2. I'm not sure if modifying argument name in a public API is a good idea. This leads to unnecessary javadoc changes.

1. I initially took the approach you suggested, but the general feedback from previous PRs (on similar work in other areas of `java.base`) has been to use more descriptive names for the pattern variables where possible. However, if you feel strongly about it please let me know which ones you would like me to change and we can discuss it further.

2. OK, I thought it might be an opportunity to add in some consistency for parameter names for the `equals()` methods. But, point taken. I've reverted these changes. Please see commit  9e9f9fb

wangweij commented on src/java.base/share/classes/java/security/cert/CertPath.java:
```diff
@@ -181,16 +181,9 @@ public boolean equals(Object other) {
-        List<? extends Certificate> otherCertList = otherCP.getCertificates();
-        return(thisCertList.equals(otherCertList));
+        return other instanceof CertPath that
+                && this.type.equals(that.getType())
```
I know `type` should never be null but let's change as little as possible by using `that.getType().equals(this.type)`.

pconcannon replied:
Ordering swapped around. See cbf2841

wangweij commented on src/java.base/share/classes/java/security/spec/EllipticCurve.java:
```diff
@@ -169,15 +169,11 @@ public BigInteger getB() {
+        return obj instanceof EllipticCurve other
+                && (field.equals(other.field)
+                && a.equals(other.a)
+                && b.equals(other.b));
```
I suppose there's no need to put the last 3 checks between "(" and ")".

pconcannon replied:
Parenthesis removed as suggested. See cbf2841

wangweij reviewed (approved):
Approved. Just some tiny comments. The other suggestion from @Punikekk also looks fine. Thanks.

pconcannon commented:
> Changes look good, except for the unneeded parenthesis?
> 
> (but I'm not a committer, so it's less useful)

Hi Jesper, thanks for your suggestions and well spotted. I've addressed the issues and responded to your comments with the commit id where the changes were made

pconcannon commented:
/integrate
